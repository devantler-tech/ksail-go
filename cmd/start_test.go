package cmd_test

import (
	"bytes"
	"testing"

	"github.com/devantler-tech/ksail-go/cmd"
)

func TestNewStartCmd(t *testing.T) {
	// Arrange
	t.Parallel()

	// Act
	cmd := cmd.NewStartCmd()

	// Assert
	if cmd == nil {
		t.Fatal("expected command to be created")
	}

	if cmd.Use != "start" {
		t.Fatalf("expected Use to be 'start', got %q", cmd.Use)
	}

	if cmd.Short != "Start a stopped Kubernetes cluster" {
		t.Fatalf("expected Short description, got %q", cmd.Short)
	}
}

func TestStartCmd_Execute(t *testing.T) {
	// Arrange
	t.Parallel()

	var out bytes.Buffer

	cmd := cmd.NewStartCmd()
	cmd.SetOut(&out)

	// Act
	err := cmd.Execute()
	// Assert
	if err != nil {
		t.Fatalf("expected no error, got %v", err)
	}

	got := out.String()

	expected := "âœ” Cluster started successfully (stub implementation)\n"

	if got != expected {
		t.Fatalf("expected output %q, got %q", expected, got)
	}
}
