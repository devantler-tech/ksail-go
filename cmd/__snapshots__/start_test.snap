
[TestStartCmdHelp - 1]
Start a previously stopped cluster.

Usage:
  start [flags]

Flags:
  -c, --context string              Kubernetes context of cluster (default "kind-ksail-default")
  -d, --distribution Distribution   Kubernetes distribution to use (default Kind)
  -h, --help                        help for start

---
